src/lib/operators/get_table.cpp

```diff
@@ -44,18 +44,45 @@ void GetTable::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeV
 
 std::shared_ptr<const Table> GetTable::_on_execute() {
   auto original_table = StorageManager::get().get_table(_name);
+
+  DebugAssert(this->transaction_context_is_set() ||
+                  [original_table]() {
+                    for (ChunkID chunk_id{0}; chunk_id < original_table->chunk_count(); ++chunk_id) {
+                      // Check, whether chunk got deleted either physically (removed from chunk-vector resp. memory).
+                      // or logically (fully invalidated)
+                      if (!original_table->get_chunk(chunk_id) ||
+                          original_table->get_chunk(chunk_id)->get_cleanup_commit_id() < MvccData::MAX_COMMIT_ID)
+                        return false;
+                    }
+                    return true;
+                  }(),
+              "GetTable needs to have a transaction context set to filter out deleted chunks.");
+
+  // Add logically & physically deleted chunks to list of excluded chunks
+  if (this->transaction_context_is_set()) {
+    for (ChunkID chunk_id{0}; chunk_id < original_table->chunk_count(); ++chunk_id) {
+      const auto chunk = original_table->get_chunk(chunk_id);
+
+      if (!chunk || chunk->get_cleanup_commit_id() <= this->transaction_context()->snapshot_commit_id()) {
+        _excluded_chunk_ids.emplace_back(chunk_id);
```

Comment:
So you would keep the set unlike [this](https://github.com/hyrise/hyrise/pull/1468/files#r253969489) comment proposed?
We could create a second vector instead of a set, which is initalized with the `_excluded_chunk_ids`, append our chunk ids and then apply sort and unique on the new vector. This could be faster than using a set.

Reply:
whatever is faster and not too verbose